povm_toolbox/library/mutually_unbiased_bases_measurements.py

```diff
@@ -0,0 +1,137 @@
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""TODO."""
+
+from __future__ import annotations
+
+import numpy as np
+from numpy.random import Generator
+from scipy.spatial.transform import Rotation
+
+from .randomized_projective_measurements import RandomizedProjectiveMeasurements
+
+
+class MutuallyUnbiasedBasesMeasurements(RandomizedProjectiveMeasurements):
+    """A mutually-unbiased-bases (MUB) POVM."""
+
+    def __init__(
+        self,
+        n_qubit: int,
+        bias: np.ndarray,
+        angles: np.ndarray,
+        measurement_twirl: bool = False,
+        measurement_layout: list[int] | None = None,  # TODO: add | Layout
+        shot_repetitions: int = 1,
+        seed_rng: int | Generator | None = None,
+    ):
+        """Implement a mutually-unbiased-bases (MUB) POVM.
+
+        This is a special case of a :class:`.RandomizedProjectiveMeasurements` (RPM) POVM. A
+        MUB-POVM corresponds to an arbitrary rotated :class:`.LocallyBiasedClassicalShadows`
+        (LBCS) POVM. That is, the MUB-POVMs can be seen as lying between RPM- and LBCS-POVMs.
+        More precisely, the set of RPM-POVMs includes the set of MUB-POVMs which includes the
+        set of LBCS-POVMs.
+
+        Args:
+            n_qubits: the number of qubits.
+            bias: can be either 1D or 2D. If 1D, it should contain float values indicating the bias
+                for measuring in each of the PVMs. I.e., its length equals the number of PVMs (3).
+                These floats should sum to 1. If 2D, it will have a new set of biases for each
+                qubit.
+            angles: can be either 1D or 2D. If 1D, it should be of length 3 and contain float values
+                to indicate the three Euler angles to rotate the locally-biased classical shadows (LBCS)
+                measurement in the Bloch sphere representation. If 2D, it will have a new set of
+                angles for each qubit. The angles are expected in the order ``theta``, ``phi``, ``lam``
+                which are the parameters of the :class:`.qiskit.circuit.library.UGate` instance used to
+                rotate the LBCS measurement effects. Note that this differs from the angles expected
+                during the initialization of a :class:`.RandomizedProjectiveMeasurements` instance,
+                where the angles are expected to be pairs of angles ``(theta, phi)`` for each projective
```

Comment:
This is a good note and maybe we should include a matching explanation in the RPM docstring (i.e. explain there that the `angles` are expected to be `(theta, phi)` and explain how they work.